### PR TITLE
Unit test pull request

### PR DESCRIPTION
**Pull request by AleWaweru**
- [ ] Performed a unit test using Rspec testing library
- [ ]example{ run the test with bin/rspec spec/person_spec.rb}

### CODE REVIEW

File: spec/teacher_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        type: 'Teacher',
+        id: teacher.id,
+        name: teacher.name,
+        age: teacher.age,
+        specialization: teacher.specialization
+      }
+      expect(teacher.to_h).to eq(expected_hash)
+    end
+  end
+end
```
Comment (Qoosim) [inline spec/teacher_spec.rb:40]: - [ ] Great work so far. There's little bug here. The `require "pry"` on top of the main file `teacher.rb` is making the tests in this file to fail. It passed when I commented out that require. Take a look.

### Test Failed
teacter-test-failed

### Test Passed
pry

Kindly attend to it and fix.

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Passed some tests. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/teacher_spec.rb (hunk 1 above)
Comment (AleWaweru) [reply spec/teacher_spec.rb:40]: done

**Review by joel-kalema — APPROVED** [review]
Hi Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights
Tests are passing 👍 
Screenshot 2023-08-11 145838

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**